pkg/rootless/rootless_linux.c

```diff
@@ -117,6 +121,143 @@ rootless_gid ()
   return rootless_gid_init;
 }
 
+/* exec the specified executable and exit if it fails.  */
+static void
+exec_binary (const char *path, char **argv, int argc)
+{
+  int r, status = 0;
+  pid_t pid;
+
+  pid = fork();
+  if (pid < 0)
+    {
+      fprintf(stderr, "fork: %m\n");
+      exit(EXIT_FAILURE);
+    }
+  if (pid == 0)
+    {
+      size_t i;
+      char **newargv = malloc((argc + 2) * sizeof(char *));
+      if (!newargv)
+        {
+          fprintf(stderr, "malloc: %m\n");
+          exit(EXIT_FAILURE);
+        }
+      newargv[0] = (char*) path;
+      for (i = 0; i < argc; i++)
+        newargv[i+1] = argv[i];
+
+      newargv[i+1] = NULL;
+      errno = 0;
+      execv(path, newargv);
+      /* If the file was deleted in the meanwhile, return success.  */
+      if (errno == ENOENT)
+        exit(EXIT_SUCCESS);
+      exit(EXIT_FAILURE);
+    }
+
+  r = TEMP_FAILURE_RETRY (waitpid (pid, &status, 0));
+  if (r < 0)
+    {
+      fprintf(stderr, "waitpid: %m\n");
+      exit(EXIT_FAILURE);
+    }
+  if (WIFEXITED(status) && WEXITSTATUS(status))
+    exit(WEXITSTATUS(status));
```

Comment:
In case of a error can we print a extra line such as `Error: external auth script failed`? If someone ever complains that podman is not working because some auth executable exits silently, it should be clear that this was not a podman error without running strace.